docs/topic-guides/workflows.rst

```diff
@@ -0,0 +1,100 @@
+.. _workflows:
+
+Renku Workflows
+===============
+
+Together with :ref:`provenance of results <provenance>`, Renku workflows is
+a key feature of Renku to make code and processing pipelines reusable.
+When tracking executions with Renku, each execution creates a workflow step.
+These steps can be joined together into bigger workflows, making it easy to
+run them with varying inputs and on different execution backends.
+
+There are many uses for this. You could define and test a pipeline locally
+and then execute it on an HPC cluster on a big dataset once you're happy.
+Or you could run your workflow offer a range of different parameters to see
+how the results differ. Or you create a workflow that runs daily with a
+CRON job.
+
+All executions of workflows are also tracked in the provenance, without you
+having to re-type and memorize the commands used in each step.
```

Comment:
```suggestion
Workflows show how code and data in a project are connected. By encoding
these relationships, your project is easier for you to manage, and faster for
others to read and reuse!

Each time to track an execution with Renku, you create a workflow step.
Encoding a workflow step makes it easier for you to rerun it without retyping
long commands. Recording workflow steps in Renku also records
:ref:`provenance of results <provenance>`, metadata that you and others
can use to understand how a data file was generated.

To take full advantage of workflows, join individual steps together into multi-step
workflows. When your code pipeline is encoded as a workflow, you can
easily re-run all or portions of your workflow with simple commands, test
your code with different parameters and compare the results, and send it
to different execution backends. 

In addition, structuring your project code as a workflow makes it easy for
someone else to understand and get up to speed with your project. No
more reading through multiple files to understand how they are connected -
workflows make the connections between code and data files easy to
understand by listing each workflow step and its inputs and outputs. 
```

Reply:
I forgot to merge this in initially but now added it separately.